### DISCUSSION TITLE
Flows run by Newman or terminal tool

### DISCUSSION THREAD

**Discussion by KamilGrabiec** [postmanlabs/postman-flows#19] [Ideas]
Hi, 
First of all, Flows looks great to me and I would like to use it with our CI/CD as we did in past with collections and Newman.
This would allow us to keep collections clean and put all business logic into flows.
Is it possible or planned to make it like this, so flows could be executed instead of collections?

**Comment by saswatds** [comment]
@KamilGrabiec Yes, running flows in CI/CD environments, like collections, is in our Roadmap. We will share more information once we start working on it.

**Reply by jagstatboy** [reply]
This is great and much anticipated!

**Reply by tkulev** [reply]
@saswatds is there any updated for when this is planned? Flows are absolutely amazing and we will like to know when we can start investing in using them for CI/CD processes.

**Reply by saswatds** [reply]
@tkulev This is slotted to be taken up after the basic offering in the app is finalised. We are evaluating if CI/CD will also be part of the basic offering, I will let everyone know about the decision soon.

**Reply by ejerskov** [reply]
@saswatds, please consider implementing this in newman. Flows is what I've been missing in my automated test setup since I started using Postman.

**Reply by bodhihawken** [reply]
amazing! flows is fantastic for creating, interconnected tests dependent on each other
Eg. 
Create Todo
Read Todo
Complete Todo
Delete Todo

if we could get them in Newman, flows will become the most important part of my CI/CD pipeline!!!

**Reply by saswatds** [reply]
We are working internally on a spec for getting flows as a CLI tool but it is mostly not going to be part of Newman. We are not sure on the timeline for this spec as there are quite a lot of factors we need to consider here. I will let you folks know once we have the spec completed and approved, and are somewhere near implementing. Thanks a lot of the feedback.

**Reply by handsome-nium** [reply]
I would be awesome if it can be automated in the CI/CD (we use Bitbucket). Another workaround could be enable flows to be triggered on demand through the Postman API. Both options will be very valuable for our use case! -> #241 
happy to see Postman team open for feedback and new ideas!

**Reply by pablocoletto** [reply]
I totally second this idea of running flows as part of a CICD pipeline. Does anyone know if there was any update on it based on @saswatds comment last January?

Good day to all of you guys

**Reply by bitroniq** [reply]
When flows are expected to be implemented in Newman?

**Reply by chrisgraham24** [reply]
Also wondering the same. Ty!

**Reply by hayden-plentific** [reply]
@saswatds - would love to get a rough idea of the timing for this piece, even just to know if it's something we might expect in 2022

**Reply by mhernegr** [reply]
Wondering too as we're looking at using postman more and more in the company, and this thing is an easy sell i think, if i can say it works with CI :)

**Reply by Sammu1982** [reply]
Any update on this feature?

**Reply by justneedham** [reply]
Yes please

**Reply by dagober** [reply]
Any info about ETA would be highly appreciated :)

**Reply by GustavoStingelin** [reply]
Up

**Reply by hayden-plentific** [reply]
@saswatds - this remains the top item in the discussion/wish list, but still not even an ETA... is it happening?

**Reply by PospolitaNV** [reply]
Up 🙏

**Reply by amirsalarsafaei** [reply]
Up

**Comment by tengkusyafiq** [comment]
I would love an ETA also. It would be great to have this, so I don't need to use collection in my deployment pipeline. ❤️

**Comment by fr3sh** [comment]
We are one year after of the original question. Could you share some good news?

**Comment by amirsalarsafaei** [comment]
I think this would be a great feature I'm stuck using Duplication, etc for CI testing, webhook aren't enough because We have an OTP login and have to use an internal endpoint. Give us a new way to integrate and test our projects, please.
